packages/enzyme/src/ShallowWrapper.js

```diff
@@ -1411,7 +1502,15 @@ class ShallowWrapper {
       if (!isCustomComponentElement(el, adapter)) {
         throw new TypeError(`ShallowWrapper::${name}() can only be called on components`);
       }
-      return this.wrap(el, null, { ...this[OPTIONS], ...options });
+      return this.wrap(el, null, {
+        ...this[OPTIONS],
+        ...options,
+        context: {
+          ...this[OPTIONS].context,
+          ...this[ROOT][CHILD_CONTEXT],
+          ...options.context,
```

Comment:
That wouldn't quite work, since we're relying on passing `options.context` in order to combine contexts as we `dive()` through multiple levels of providers.

I've updated the code to this:
```javascript
context: options.context || {
  ...this[OPTIONS].context,
  ...this[ROOT][CHILD_CONTEXT],
}
```

So now `options.context` (passed to `dive()`) always takes precedence, but context is still combined when `dive()`ing multiple providers.

Reply:
hmm. I'm not sure that's what we want. if it's not in the child context, it shouldn't be passed to the new child.

Reply:
ahhhh ok - then yes, the larger object should indeed be passed down, as long as the component only sees the filtered context. Thanks for explaining.

Reply:
My pleasure!